### PR TITLE
Add Logging to Geolocation

### PR DESCRIPTION
**Pull request by mgupta27**
## Add Logging to Geolocation

**Key Changes**
- Added logging to the geolocation module
- Added string representation for `DetectionInWorld`

**Testing**
- All unit tests pass
- Ran `test_geolocation_worker` integration test works and confirm logs are logged as expected

Output from log file
```
06:32:52: [INFO] [C:\Users\Mihir\vscode_workspace\WARG\Repositories\computer-vision-python\modules\geolocation\geolocation.py | create | 43] geolocation logger initialized
06:32:52: [INFO] [C:\Users\Mihir\vscode_workspace\WARG\Repositories\computer-vision-python\modules\geolocation\geolocation.py | run | 323] <class 'modules.detection_in_world.DetectionInWorld'>, vertices: [[100.0, -100.00000000000001], [100.00000000000001, 100.00000000000001], [-100.0, -99.99999999999999], [-100.0, 99.99999999999999]], centre: [0. 0.], label: 1, confidence: 1.0
06:32:52: [INFO] [C:\Users\Mihir\vscode_workspace\WARG\Repositories\computer-vision-python\modules\geolocation\geolocation.py | run | 323] <class 'modules.detection_in_world.DetectionInWorld'>, vertices: [[100.0, -100.00000000000001], [100.00000000000001, 100.00000000000001], [-100.0, -99.99999999999999], [-100.0, 99.99999999999999]], centre: [0. 0.], label: 1, confidence: 1.0
06:32:52: [INFO] [C:\Users\Mihir\vscode_workspace\WARG\Repositories\computer-vision-python\modules\geolocation\geolocation.py | run | 323] <class 'modules.detection_in_world.DetectionInWorld'>, vertices: [[100.0, -100.00000000000001], [100.00000000000001, 100.00000000000001], [-100.0, -99.99999999999999], [-100.0, 99.99999999999999]], centre: [0. 0.], label: 1, confidence: 1.0
06:32:53: [INFO] [C:\Users\Mihir\vscode_workspace\WARG\Repositories\computer-vision-python\modules\geolocation\geolocation.py | run | 323] <class 'modules.detection_in_world.DetectionInWorld'>, vertices: [[100.0, -100.00000000000001], [100.00000000000001, 100.00000000000001], [-100.0, -99.99999999999999], [-100.0, 99.99999999999999]], centre: [0. 0.], label: 1, confidence: 1.0
06:32:53: [INFO] [C:\Users\Mihir\vscode_workspace\WARG\Repositories\computer-vision-python\modules\geolocation\geolocation.py | run | 323] <class 'modules.detection_in_world.DetectionInWorld'>, vertices: [[100.0, -100.00000000000001], [100.00000000000001, 100.00000000000001], [-100.0, -99.99999999999999], [-100.0, 99.99999999999999]], centre: [0. 0.], label: 1, confidence: 1.0
06:32:53: [INFO] [C:\Users\Mihir\vscode_workspace\WARG\Repositories\computer-vision-python\modules\geolocation\geolocation.py | run | 323] <class 'modules.detection_in_world.DetectionInWorld'>, vertices: [[100.0, -100.00000000000001], [100.00000000000001, 100.00000000000001], [-100.0, -99.99999999999999], [-100.0, 99.99999999999999]], centre: [0. 0.], label: 1, confidence: 1.0

```

### CODE REVIEW

File: modules/geolocation/geolocation.py [hunk 1]
```diff
@@ -158,7 +183,8 @@ def __get_perspective_transform_matrix(
         # All exceptions must be caught and logged as early as possible
         # pylint: disable-next=bare-except
         except:
-            # TODO: Logging
+            frame = inspect.currentframe()
+            self.__logger.error("could not get perspective transform matrix", frame)
             return False, None
```
Comment (TongguangZhang) [inline modules/geolocation/geolocation.py:188]:
```
except cv2.error as e:
   # log the specific cv2 error
    return False, None
except Exception as e:
    # log the specific unexpected error type
    return False, None
```

File: modules/geolocation/geolocation.py [hunk 2]
```diff
@@ -45,6 +57,10 @@ def create(
             # Image space to camera space
             result, value = camera_intrinsics.camera_space_from_image_space(source[0], source[1])
             if not result:
+                frame = inspect.currentframe()
+                geolocation_logger.error(
+                    f"rotated source vector could not be created for source: {source}", frame
```
Comment (Xierumeng) [inline modules/geolocation/geolocation.py:62]: Capitalize. Same for logging statements below.

File: modules/geolocation/geolocation.py [hunk 3]
```diff
@@ -285,7 +315,13 @@ def run(
...
             # Partial data not allowed
             if not result:
+                frame = inspect.currentframe()
+                self.__logger("could not convert detection to world from image", frame)
                 return False, None
             detections_in_world.append(detection_world)
 
+        for detection_in_world in detections_in_world:
+            frame = inspect.currentframe()
+            self.__logger.info(detection_in_world, frame)
```
Comment (TongguangZhang) [inline modules/geolocation/geolocation.py:325]: we should log each one as it's created in the for loop above (after we append it to the list

**Review by TongguangZhang — COMMENTED** [review]
reviewed

**Review by Xierumeng — DISMISSED** [review]
Conditionally approved.

File: modules/geolocation/geolocation.py (hunk 2 above)
Comment (mgupta27) [reply modules/geolocation/geolocation.py:62]: Sounds good. I noticed in main the logger outpus `main logger initialized` which is why I did all lowercase. Should those be changed as well?

**Pull request closed without merging**